x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/cluster/routing/allocation/DataTierAllocationDecider.java

```diff
@@ -91,10 +91,17 @@ public void validate(String value) {
 
         @Override
         public void validate(String value, Map<Setting<?>, Object> settings) {
-            if (Strings.hasText(value) && SearchableSnapshotsConstants.isPartialSearchableSnapshotIndex(settings) == false) {
-                String[] split = value.split(",");
-                if (Arrays.stream(split).anyMatch(DATA_FROZEN::equals)) {
-                    throw new IllegalArgumentException("[" + DATA_FROZEN + "] tier can only be used for partial searchable snapshots");
+            if (Strings.hasText(value)) {
```

Comment:
I think we need to either make `_tier_preference` validate that partial searchable snapshots always have the value `data_frozen` (i.e., must be set) or else make that the default when unset.

Ideally, we should remove support for include/exclude/require now that we have the chance too? Can be a separate PR if that is easier.

Reply:
> Ideally, we should remove support for include/exclude/require now that we have the chance too? Can be a separate PR if that is easier.

Are you talking about include/exclude/require for `_tier`, or *all* include/exclude/require parameters? If you mean only for tier, I think I'd prefer to work out how we want to treat them for regular indices first (since we talked about removing `_tier` in favor of `_tier_preference` for *all* indices) before special casing them for only frozen indices

Reply:
I was only referring to include/exclude/require for `_tier`. We can tackle it outside this PR. But it feels odd that the only legal exclude value is `data_frozen` for frozen indices, since that is the last thing you want. When you modify `_tier_preference` validation to require `data_frozen`, the validation have to be different for include/exclude/require vs `_tier_preference` anyway, which is why I was suggesting to tackle this now. But perfectly fine to tackle later too.

Reply:
> I think we need to either make _tier_preference validate that partial searchable snapshots always have the value data_frozen (i.e., must be set) or else make that the default when unset.

I pushed 9bab290fcd7a1b42fba366c331bd44344899d37f for this

Reply:
I wonder if there is still a loophole of setting `_tier_preference` to either `""` (empty) or `" "` (white-space), since then the default value will not apply and we end up with a frozen shard with no `_tier_preference`? I think we need to just check for null here to fix that, at least for frozen.

Reply:
Yep, I changed this to be a null comparison